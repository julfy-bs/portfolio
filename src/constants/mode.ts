/**
 * Значение класса для применения стилей темной темы.
 * @memberof Mode
 * @alias ModeEnumClass
 * @readonly
 * @enum {'page_theme_dark'} DarkMode
 */
export const MODE_DARK_CLASS = 'page_theme_dark';
/**
 * Значение переменной-ключа локального хранилища (local storage) в котором хранится текущая тема пользователя.
 * @memberof Mode
 * @alias ModeEnumStorageKey
 * @readonly
 * @enum {'isDarkMode'} DarkMode
 */
export const MODE_LOCAL_STORAGE_KEY = 'isDarkMode';

/**
 * Значения переменных для смены темы.
 * @memberof Mode
 * @alias ModeEnum
 * @readonly
 * @enum {object} DarkMode
 */
export enum DarkMode {
  /**
   * ENUM значение для светлой темы.
   * @readonly
   * @enum {string}
   * */
  FALSE = '0',
  /**
   * ENUM значение для случая, когда тема отсутствует.
   * @readonly
   * @enum {string}
   * */
  NULL = '1',
  /**
   * ENUM значение для темной темы.
   * @readonly
   * @enum {string}
   * */
  TRUE = '2',
}
