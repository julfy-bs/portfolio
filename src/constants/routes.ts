export const PATH = {
  HOME: '/',
  LOGIN: '/login',
  PROFILE: '/profile',
};
