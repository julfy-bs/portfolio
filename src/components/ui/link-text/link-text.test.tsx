import { render, screen } from '@testing-library/react';
import { FC, ReactNode } from 'react';
import { BrowserRouter, Route, Routes } from 'react-router-dom';
import { describe, expect, test } from 'vitest';

import { PATH } from '../../../constants/routes';
import { LINK_TEST_ID } from '../../../constants/tests/text-link/values';
import LinkText from './link-text';

type ElementToTestProps = {
  children?: ReactNode;
  external?: boolean;
  navigation?: boolean;
};
const testingLinkText = 'TEST';
const linkHrefAttribute = 'href';
const testingRoute = PATH.HOME;
const ElementToTest: FC<ElementToTestProps> = ({ children, navigation = false, external = false }) => {
  return (
    <BrowserRouter>
      <Routes>
        <Route
          path={testingRoute}
          element={
            <LinkText text={testingLinkText} route={testingRoute} navigation={navigation} external={external}>
              {children}
            </LinkText>
          }
        />
      </Routes>
    </BrowserRouter>
  );
};

describe('UI-компонент <LinkText /> рендерится без ошибок', () => {
  test('С обязательными параметрами.', () => {
    render(<ElementToTest />);
    const link: HTMLButtonElement = screen.getByTestId(LINK_TEST_ID);
    expect(link).toHaveTextContent(testingLinkText);
    expect(link.classList.contains('link'));
    expect(link.getAttribute(linkHrefAttribute)).toBe(testingRoute);
    expect(link).toBeInTheDocument();
    expect(link).toMatchSnapshot();
  });

  test('Рендерится с параметром navigation', () => {
    render(<ElementToTest navigation={true} />);
    const link: HTMLButtonElement = screen.getByTestId(LINK_TEST_ID);
    expect(link.classList.contains('link_role_navigation'));
    expect(link.classList.contains('link'));
    expect(link).toHaveTextContent(testingLinkText);
    expect(link.getAttribute(linkHrefAttribute)).toBe(testingRoute);
    expect(link).toBeInTheDocument();
    expect(link).toMatchSnapshot();
  });
});
