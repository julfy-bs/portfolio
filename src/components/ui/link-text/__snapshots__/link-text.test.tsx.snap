// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`UI-компонент <LinkText /> рендерится без ошибок > Рендерится с параметром навигации 1`] = `
<a
  aria-current="page"
  class="_link_dd42dc _link_active_dd42dc _link_role_navigation_dd42dc"
  data-testid="link-element"
  href="/"
  tabindex="-1"
>
  NaveLink Test
  <span
    class="_text_dd42dc"
    data-testid="link-text"
  >
    Главная
  </span>
</a>
`;

exports[`UI-компонент <LinkText /> рендерится без ошибок > С обязательными параметрами. 1`] = `
<a
  aria-current="page"
  class="_link_dd42dc _link_active_dd42dc"
  data-testid="link-element"
  href="/"
  tabindex="-1"
>
  <span
    class="_text_dd42dc"
    data-testid="link-text"
  >
    TEST
  </span>
</a>
`;
